schemas/input/csl-data.json

```
@@ -437,6 +437,110 @@
     "additionalProperties": false
   },
   "definitions": {
+    "edtf-string": {
+      "title": "EDTF date string",
+      "description": "CSL input supports EDTF, levels 0 and 1.",
+      "type": "string",
+      "pattern": "^[0-9-%~X?./]{4,}$"
+    },
+    "date-object": {
+      "title": "Date Object",
+      "description": "An EDTF date represented as an object. Seasons can be represented using the 21-24 month values of EDTF, decades and centuries using the X notation, and the qualifiers with the included boolean properties",
+      "examples": [
+        {
+          "id": "one",
+          "type": "book",
+          "title": "The Title",
+          "issued": {
+            "date-parts": [2000, 3, 10]
+          },
+          "approximate": "true"
+        }
+      ],
+      "type": "object",
+      "properties": {
+        "year": {
+          "title": "year",
+          "description": "A year; can be negative.",
+          "examples": ["2012","-32"],
+          "type": "integer"
+        },
+        "month": {
+          "title": "month",
+          "description": "A month; can also represent seasons using the EDTF 21-24 values.",
+          "examples": ["2","22"],
+          "type": "integer"
+        },
+        "day": {
+          "title": "day",
+          "type": "integer"
+        },
+        "literal": {
+          "title": "Literal date",
+          "description": "A date that should be passed through to the processor as is; for dates that cannot be represented in EDTF.",
+          "examples": ["Han Dynasty"],
+          "type": "string"
+        },
+        "approximate": {
+          "title": "Approximate",
+          "description": "Indicates an approximate or circa date, which may be presented as `circa May 23, 1972`.",
+          "type": "boolean"
+        },
+        "uncertain": {
+          "title": "Uncertain",
+          "description": "Indicates uncertainty about a date representation, which may be presented as `May 23, 1972?`",
+          "type": "boolean"
+        }
+      }
+    },
+    "date-range": {
+      "title": "Date Range",
+      "description": "An EDTF range is a two-item array. An open end or beginning of a range can be represented with an empty (date) object.",
+      "type": "array",
+      "examples": [
+        {
+          "id": "range-1",
+          "type": "book",
+          "title": "The Title",
+          "issued": [{ "date-parts": [2000] }, { "date-parts": [2001] }]
```

If we ditch date-parts:

```suggestion
          "issued": [{ "year": "2000" }, { "year": "2001" }]
```